source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -366,7 +366,7 @@ void CacheFilter::processSuccessfulValidation(Http::ResponseHeaderMap& response_
   // freshly served response from the origin, unless the 304 response has an Age header, which
   // means it was served by an upstream cache.
   // Remove any existing Age header in the cached response.
-  lookup_result_->headers_->removeInline(age_handle.handle());
+  lookup_result_->headers_->removeInline(CacheCustomHeaders::get().age.handle());
```

Comment:
suggest capturing CacheCustomHeaders::get() in a local temp to avoid multiple locked lookups of the static within the function.

Reply:
I addressed this via the previous suggestion to exchange get() for a method per header.